checker.cpp

```diff
@@ -1,22 +1,294 @@
+    }
+
+    bool checkChargeRateIsInRange(float chargeRate)
+    {
+        if (chargeRate >= 0.8)
+            return false;
+        else
+            return NORMAL;
+    }
+
+    bool valueIsInRange(float value, int min, int max)
+    {
+        if (value < min || value > max)
+            return false;
+        else
+            return true;
+    }
+
+    bool checkBatteryIsOk(float temp, float soc, float chargerate)
+    {
+        return(checkTempIsInRange(temp, mintemp, maxtemp) && checkSocIsInRange(soc, minsoc, maxsoc) && checkChargeRateIsInRange(chargerate));
+    }
+
+    bool checkTempIsInWarningRange(float temp, char* lang)
+    {
+        int warninglevel = valuesIsInWarningRange(temp, mintemp, maxtemp, lowlimtempwarn, highlimtempwarn);
+        switch (warninglevel)
+        {
+        case 1:
+            LogLowTempWarning(lang);
+            return true;
+        case 2:
+            LogHighTempWarning(lang);
+            return true;
+        }
+        return false;
+    }
+
+    bool checkSocIsInWarningRange(float soc, char* lang)
+    {
+        int warninglevel = valuesIsInWarningRange(soc, minsoc, maxsoc, lowlimsocwarn, highlimsocwarn);
+        switch (warninglevel)
+        {
+        case 1:
+            LogLowSocWarning(lang);
+            return true;
+        case 2:
+            LogHighSocWarning(lang);
+            return true;
+        }
+        return false;
+    }
+
+    bool checkChargeRateIsInWarningRange(float chargerate, char* lang)
+    {
+        if (0.8 > chargerate > chargeratewarn)
+        {
+            LogChargeRateWarning(lang);
+            return true;
+        }
+        return false;
+    }
+
+    void LogHighTempWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: Approaching High Temperature Range\n";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung: Annäherung an den hohen Temperaturbereich\n";
+        }
+    }
+
+    void LogLowTempWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: Approaching Low Temperature Range\n";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung: Annäherung an den niedrigen Temperaturbereich\n";
+        }
+    }
+
+    void LogHighSocWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: Approaching charge-peak\n";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung: Ladespitze nähert sich\n";
+        }
+    }
+
+    void LogLowSocWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: Approaching discharge\n";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung:  Naht Entladung\n";
+        }
+    }
+
+    void LogChargeRateWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
```

Comment:
every new addition of language will require you to modify this and similar code in the project. that would lead to lot of rework and could be error prone,
One possible solution could be to maintain a list of strings (i.e. a list of English strings, a list of German strings) in a separate ,h file and maybe at compile time of run time select appropriate list (could be based on a hardcoded global value to keep it simple as of now).